components/scream/src/mct_coupling/scream_scorpio_interface.cpp

```diff
@@ -0,0 +1,118 @@
+#include "scream_scorpio_interface.hpp"
+#include "scream_config.h"
+
+#include "ekat/scream_assert.hpp"
+#include "ekat/util/scream_utils.hpp"
+#include "ekat/scream_types.hpp"
+
+#include <string>
+
+using scream::Real;
+using scream::Int;
+extern "C" {
+
+// Fortran routines to be called from C++
+  void grid_write_data_array_c_real_1d(const std::string &filename, const std::string &varname, const Int dim1_length, const Real* hbuf);
```

Comment:
I think these have to be changed. Passing an std::string to F90 is not safe. It relies on the fact that the actual char* is the first thing stored in the object. That is currently the case, but it is quite a subtle assumption. If someone came out with an impl of the std library where the first thing stored was the string length, nasty bugs would ensue.

It is quite easy, however, to make these safe. Simply require a `const char*&` as input arg, and invoke the function as `grid_blah (filename.c_str(),...)`.